nilearn/regions/region_extractor.py

```diff
@@ -0,0 +1,320 @@
+"""
+Better brain parcellations for Region of Interest analysis
+"""
+import warnings
+import numbers
+import nibabel
+import numpy as np
+
+from scipy.ndimage import label
+
+from sklearn.base import clone
+from sklearn.externals.joblib import Memory
+
+from .. import masking
+from ..input_data import NiftiMapsMasker
+from .._utils import check_niimg, check_niimg_4d
+from ..image import new_img_like, resample_img
+from ..image.image import _smooth_array, threshold_img
+from .._utils.niimg_conversions import concat_niimgs, _check_same_fov
+from .._utils.compat import _basestring
+from .._utils.ndimage import _peak_local_max
+from .._utils.segmentation import _random_walker
+
+
+def _threshold_maps(maps, threshold):
+    """ Automatic threshold function using ratio to the number of voxels
+    in the brain volume to keep the most prominent features of the maps
+    which lies above a certain threshold
+    """
+    maps = check_niimg(maps)
+    maps_data = maps.get_data()
+    abs_maps = np.abs(maps_data)
+    if isinstance(threshold, float):
```

Comment:
Also do this sanity check towards the top of the function. No need to do .get_data if threshold is not a float.